File: year2018/day2/python/inventory.py
def n_occurances(items, count):
    total = 0
    for id in items:
        occurances = [id.count(char) == count for char in id]
        total += True in occurances
    return total

if __name__ == '__main__':
    with open('input.txt', 'r') as f:
        items = [line for line in f]
    print(f"Checksum: {n_occurances(items, 2) * n_occurances(items, 3)}")

File: year2018/day2/C/inventory.c
#include <stdio.h>
#include <stdlib.h>

int find_n_occurrences(char* string, int occurrences){
    int count = 0;
    for(int i = 0; string[i] != '\0'; i++){
        for(int j = 0; string[j] != '\0'; j++){
            count += string[j] == string[i];
        }
        if (count == occurrences){
            return 1;
        }
        count = 0;
    }
    return 0;
}

int main(){
    char input[30];
    int two_count = 0, three_count = 0;
    FILE* fp = fopen("input.txt", "r");
    if(fp == NULL){
        printf("Error. Could not open file!");
        exit(-1);
    }
    while(fscanf(fp, "%s", &input) != EOF){
        two_count += find_n_occurrences(input, 2);
        three_count += find_n_occurrences(input, 3);
    }
    printf("Checksum: %d", two_count * three_count);
    return 0;
}

